File: LeetCode/401-500/448. Find All Numbers Disappeared in an Array/Solution.cpp
class Solution {
public:
  vector<int> findDisappearedNumbers(vector<int> &nums) {
    for (int i = 0; i < nums.size(); i++) {
      int ab = abs(nums[i]);
      nums[ab - 1] = abs(nums[ab - 1]) * -1;
    }

    vector<int> res;
    for (int c = 0; c < nums.size(); c++) {
      if (nums[c] > 0) {
        res.push_back(c + 1);
      }
    }

    return res;
  }
};

File: LeetCode/401-500/448. Find All Numbers Disappeared in an Array/Solution.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            while (nums[i] != i + 1 && nums[i] != nums[nums[i] - 1]) {
                int tmp = nums[i];
                nums[i] = nums[tmp - 1];
                nums[tmp - 1] = tmp;
            }
        }

        List<Integer> list = new ArrayList<Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != i + 1)
                list.add(i + 1);
        }
        return list;
    }
}